### DISCUSSION TITLE
This extension saved my dependency hell and I would like to understand why

### DISCUSSION THREAD

**Discussion by andrsbtrg** [Nice3point/RevitExtensions#5] [Q&A] [answered]
Hello,

I was chasing a bug for some time, where when I opened a command I would get an system exception where a dependency was not found. The dependency was System.Text.Json version 7.0.0. I noticed this happened only on a command which calls an external library "Airtable" which depends upon System.Text.Json version 7.0.2. I did everything I could to solve this, I installed the exact version from Nuget, I researched the assembly binding logs, etc.
Then I noticed that I also call the Airtable library from another command, but on that command there was no error. Furthermore, if I call the second command first, and then I try to run the first command again, there is no error.

Finally I realised that the second command I made inherits from `ExternalCommand` from Nice3point.Revit.Toolkit.External, but the first command was implementing `IExternalCommand` interface directly.

The fix to make the first command work was to inherit from `ExternalCommand` as well. I am really glad this extensions exist, but I really wonder why does using `ExternalCommand` bind correctly the dependency, which doesn't happen with the RevitApi `IExternalCommand` interface?

**Comment by Nice3point** [comment, marked as answer]
Hi, this issue related to the https://github.com/Nice3point/RevitToolkit, not RevitExtensions 

nevertheless, I'll try to explain what this is all about

**ExternalCommand** used `ResolveHelper` https://github.com/Nice3point/RevitToolkit/blob/main/source/Nice3point.Revit.Toolkit/Helpers/ResolveHelper.cs under the hood for Revit2020-2024 and `AssemblyLoadContext` for Revit2025+ https://github.com/Nice3point/RevitToolkit/blob/main/source/Nice3point.Revit.Toolkit/Helpers/AddinLoadContext.cs 

This info is described in the Readme https://github.com/Nice3point/RevitToolkit?tab=readme-ov-file#resolvehelper

All this helps to resolve dependencies, and load the correct libraries. In the empty `IExternalCommand` interface there is no such possibility, that's why you had problems with libraries

**Reply by andrsbtrg** [reply]
Ah that's exactly what I was looking for, next time I will look further down the readme. Thank you so much for the answer and developing this awesome libraries
